### PR TITLE
test: refactor UserSignupController test

### PR DESCRIPTION
**Pull request by xcoulon**
use more MasterUserRecordAssertion from toolchain-common
(see see https://github.com/codeready-toolchain/toolchain-common/pull/229)

also, remove duplicate imports

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: controllers/usersignup/usersignup_controller_test.go [hunk 1]
```diff
@@ -101,27 +101,13 @@ func TestUserSignupCreateMUROk(t *testing.T) {
...
-					TemplateRef: "base-dev-123abc1",
-				},
-				{
-					TemplateRef: "base-stage-123abc2",
-				},
-			}, mur.Spec.UserAccounts[0].Spec.NSTemplateSet.Namespaces)
-			require.NotNil(t, mur.Spec.UserAccounts[0].Spec.NSTemplateSet.ClusterResources)
-			assert.Equal(t, "base-clusterresources-654321b", mur.Spec.UserAccounts[0].Spec.NSTemplateSet.ClusterResources.TemplateRef)
-
+			murtest.AssertThatMasterUserRecord(t, "foo", r.Client).
```
Comment (MatousJobanek) [inline controllers/usersignup/usersignup_controller_test.go:104]: to properly replace the existing test code, we should list the MURs and expect that the number is `1`, plus do the additional checks...
the same applies also to the other cases

Comment (xcoulon) [reply controllers/usersignup/usersignup_controller_test.go:104]: just to clarify: I believe I transposed all the additional checks, but maybe I forgot something? And sure, I can check that there's only 1 MasterUserRecord in each test

File: controllers/usersignup/usersignup_controller_test.go [hunk 2]
```diff
@@ -1788,12 +1731,9 @@ func TestUserSignupDeactivatedAfterMURCreated(t *testing.T) {
...
 			})
 
-		murs := &toolchainv1alpha1.MasterUserRecordList{}
-
 		// The MUR should have now been deleted
-		err = r.Client.List(context.TODO(), murs)
-		require.NoError(t, err)
-		require.Len(t, murs.Items, 0)
+		murtest.AssertThatMasterUserRecords(t, r.Client).HaveCount(0)
+		murtest.AssertThatMasterUserRecord(t, "john-doe", r.Client).DoesNotExist()
```
Comment (MatousJobanek) [inline controllers/usersignup/usersignup_controller_test.go:1736]: btw, do we still need the:
```go
murtest.AssertThatMasterUserRecord(t, "john-doe", r.Client).DoesNotExist()
```
when we already checked that the list of the MURs is 0?
```suggestion
		murtest.AssertThatMasterUserRecords(t, r.Client).HaveCount(0)
```
the same applies also to all following cases.

File: go.mod [hunk 3]
```diff
@@ -31,4 +31,6 @@ require (
 	sigs.k8s.io/controller-runtime v0.8.3
 )
 
+replace github.com/codeready-toolchain/toolchain-common => github.com/xcoulon/toolchain-common v0.0.0-20220110151033-c6fb7af95fd2
```
Comment (MatousJobanek) [inline go.mod:34]: reminder

**Review by MatousJobanek — APPROVED** [review]
:+1:

File: controllers/usersignup/usersignup_controller_test.go (hunk 2 above)
Comment (xcoulon) [reply controllers/usersignup/usersignup_controller_test.go:1736]: yeah, you're right. I probably made those latest changes too quickly did not pay attention to these cases 🤷‍♂️

Comment (xcoulon) [reply controllers/usersignup/usersignup_controller_test.go]: done in 12c4184

File: go.mod (hunk 3 above)
Comment (xcoulon) [reply go.mod]: addressed in 12c4184

**Pull request merged**